### PR TITLE
add support for base path in docker cli -H

### LINKED ISSUE #14270: Proposal: Support base url in the client
Today in the client you can use `-H` with `unix://<file>`, `tcp://<ip:port>` and `fd://<num>`

I'm currently hosting my docker daemon behind a Reverse Proxy (nginx).
I wonder if the docker client should support

`-H tcp://<ip:port></path>` or `-H http://<ip></path>` 

What do you guys think ?

### PR DESCRIPTION
**Pull request by vieux**
Close #14270

### CODE REVIEW

**Comment by LK4D4** [comment]
@vieux rebase needed :)

**Comment by LK4D4** [comment]
Also I'm pretty sure that docs update needed too.

**Comment by vieux** [comment]
@LK4D4 done :D

**Comment by vieux** [comment]
ping @tnachen @jimenez

**Comment by jimenez** [comment]
:+1:

**Comment by tnachen** [comment]
LGTM

**Comment by runcom** [comment]
+1

**Comment by vdemeester** [comment]
@vieux I think it will close #13871 then.

File: docs/articles/basics.md [hunk 1]
```diff
@@ -83,12 +83,14 @@ Similarly, the Docker client can use `-H` to connect to a custom port.
 
 `-H` accepts host and port assignment in the following format:
 
-    tcp://[host][:port] or unix://path
+    tcp://[host][:port][path] or unix://path
```
Comment (LK4D4) [inline docs/articles/basics.md]: Shouldn't here be `/` between `[:port]` and `[path]`?

**Comment by LK4D4** [comment]
Design is ok for me
ping @docker/core-maintainers

File: docs/articles/basics.md (hunk 1 above)
Comment (vieux) [inline docs/articles/basics.md]: @LK4D4 to me the / is included in the path. I can use ` tcp://[host][:port][/path]` if you'd like

**Comment by icecrime** [comment]
+1

**Comment by icecrime** [comment]
LGTM (needs rebase though)

**Comment by vieux** [comment]
@LK4D4 @icecrime rebased.

**Comment by jfrazelle** [comment]
LGTM, just waitinng on janky to pass

**Pull request merged**